common/src/test/java/org/opensearch/ml/common/utils/StringUtilsTest.java

```diff
@@ -457,4 +458,35 @@ public void testGetJsonPath_ValidJsonPathWithoutSource() {
         String result = getJsonPath(input);
         assertEquals("$.response.body.data[*].embedding", result);
     }
+
+    @Test
+    public void testisValidJSONPath_ValidInputs() {
```

Comment:
As I provided more complex examples, can we add those in the test too?

Reply:
you mean these ones?
$.store['book']
$['store']['book'][0]['title']
Array Indexing:

$.store.book[0]
$.store.book[1,2] (Multiple indices)
$.store.book[-1:] (The last element)
$.store.book[0:2] (Slice array from index 0 to 2)
Wildcard (*):

$.store.book[*] (All books)
$.* (All properties at the root level)
Filters (?()):

$.store.book[?(@.price < 10)] (Books cheaper than 10)
$.store.book[?(@.author == 'J.K. Rowling')] (Books authored by J.K. Rowling)
Recursive Descent (..):

$..author (Find all authors, no matter where they are in the document)
$..book[?(@.price > 15)] (Find books with a price greater than 15 anywhere in the document)
Union ([,]):

$.store.book[0,1] (First and second books)
$['store','warehouse'] (Select multiple properties at the same level)
Combination of Wildcard and Filters:

$..book[?(@.price > 20)].title (Get the titles of books costing more than 20)

Reply:
Yeah, if you think some of them could be related to your use case?

Reply:
sounds great, added in
[add more tests](https://github.com/opensearch-project/ml-commons/pull/2899/commits/576c3fa7f60591299aac494adc991fef84b9d20b)